### PR TITLE
Feature/solana migration

### PR DESCRIPTION
**Pull request by jakcinmarina**

### CODE REVIEW

File: packages/blockchain/solana/src/lib/solana.sdk.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+export const TatumSolanaSDK = (args: SDKArguments) => {
+  const api = BlockchainSolanaService
+  const web3: SolanaWeb3 = solanaWeb3()
+
+  return {
+    ...abstractBlockchainSdk({
+      ...args,
+      blockchain,
+    }),
+    api,
```
Comment (npwork) [inline packages/blockchain/solana/src/lib/solana.sdk.ts:20]: please remove api

File: packages/blockchain/solana/src/lib/solana.sdk.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    httpDriver: async (request: Web3Request): Promise<Web3Response> => {
+      return api.solanaWeb3Driver(args.apiKey, { ...request, jsonrpc: '2.0' })
+    },
+    kms: solanaKmsService({ web3 }),
+    transaction: solanaTxService({ web3 }),
+    blockchain: {
+      getCurrentBlock: BlockchainSolanaService.solanaGetCurrentBlock,
+      getBlock: BlockchainSolanaService.solanaGetBlock,
+      getAccountBalance: BlockchainSolanaService.solanaGetBalance,
+      getTransaction: BlockchainSolanaService.solanaGetTransaction,
```
Comment (npwork) [inline packages/blockchain/solana/src/lib/solana.sdk.ts:31]: I think there are more functions available in BlockchainSolanaService

File: packages/blockchain/solana/src/lib/solana.sdk.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      blockchain,
+    }),
+    api,
+    wallet: BlockchainSolanaService.solanaGenerateWallet,
+    httpDriver: async (request: Web3Request): Promise<Web3Response> => {
+      return api.solanaWeb3Driver(args.apiKey, { ...request, jsonrpc: '2.0' })
+    },
+    kms: solanaKmsService({ web3 }),
+    transaction: solanaTxService({ web3 }),
+    blockchain: {
```
Comment (npwork) [inline packages/blockchain/solana/src/lib/solana.sdk.ts:27]: please add tests for mapping (check btc as an example)

**Review by vlnevyhosteny — CHANGES_REQUESTED** [review]

File: packages/blockchain/solana/src/lib/schema/schema.ts [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    this.updateAuthority = args.updateAuthority
+    this.mint = args.mint
+    this.data = args.data
+    this.primarySaleHappened = args.primarySaleHappened
+    this.isMutable = args.isMutable
+    this.editionNonce = args.editionNonce ?? null
+  }
+}
+
+export const METADATA_SCHEMA = new Map<any, any>([
```
[the remark this answers is not in the document]
Comment (jakcinmarina) [reply packages/blockchain/solana/src/lib/schema/schema.ts:127]: let's fix after release

**Review by vlnevyhosteny — APPROVED** [review]

**Comment by npwork** [comment]
@jakcinmarina  please resolve conflicts

**Comment by jakcinmarina** [comment]
updated

**Pull request merged**